atlasdb-config/src/main/java/com/palantir/atlasdb/http/FeignOkHttpClients.java

```diff
@@ -105,14 +105,15 @@ private static Client newOkHttpClient(
             Optional<SSLSocketFactory> sslSocketFactory,
             String userAgent,
             boolean retryOnConnectionFailure) {
-        com.squareup.okhttp.OkHttpClient client = new com.squareup.okhttp.OkHttpClient();
-
-        client.setConnectionSpecs(CONNECTION_SPEC_WITH_CYPHER_SUITES);
-        client.setConnectionPool(new ConnectionPool(CONNECTION_POOL_SIZE, KEEP_ALIVE_TIME_MILLIS));
-        client.setSslSocketFactory(sslSocketFactory.orNull());
-        client.setRetryOnConnectionFailure(retryOnConnectionFailure);
-        client.interceptors().add(new UserAgentAddingInterceptor(userAgent));
-        return new OkHttpClient(client);
+        okhttp3.OkHttpClient.Builder builder = new okhttp3.OkHttpClient.Builder()
+                .connectionSpecs(CONNECTION_SPEC_WITH_CYPHER_SUITES)
+                .connectionPool(new ConnectionPool(CONNECTION_POOL_SIZE, KEEP_ALIVE_TIME_MILLIS, TimeUnit.MILLISECONDS))
+                .retryOnConnectionFailure(retryOnConnectionFailure);
+        if (sslSocketFactory.isPresent()) {
+            builder.sslSocketFactory(sslSocketFactory.get());
```

Comment:
The way we were using sslSocketFactory before is now deprecated
(Judging from the impl, they deprecated it to try and get rid of some kind of gross reflection code)

They actually want you to pass in a (sslSocketFactory, X509TrustManager) here, now.
They list some example code in the javadoc (because building one of these requires some puke-worthy java)
https://square.github.io/okhttp/3.x/okhttp/okhttp3/OkHttpClient.Builder.html#sslSocketFactory-javax.net.ssl.SSLSocketFactory-
```
   TrustManagerFactory trustManagerFactory = TrustManagerFactory.getInstance(
       TrustManagerFactory.getDefaultAlgorithm());
   trustManagerFactory.init((KeyStore) null);
   TrustManager[] trustManagers = trustManagerFactory.getTrustManagers();
   if (trustManagers.length != 1 || !(trustManagers[0] instanceof X509TrustManager)) {
     throw new IllegalStateException("Unexpected default trust managers:"
         + Arrays.toString(trustManagers));
   }
   X509TrustManager trustManager = (X509TrustManager) trustManagers[0];
```

Do I need to pipe the Atlas Config all the way into here so I can get our keystore location from it? Why was this not required before for SSL to work?

@schlosna maybe would know better how about this sort of thing?